include/deal.II/rol/vector_adaptor.h

```diff
@@ -0,0 +1,478 @@
+   * VectorType::operator+=(const VectorType &);
+   *
+   *                       // Scale all elements by a given scalar.
+   * VectorType::operator*=(const VectorType::value_type &);
+   *
+   *                       // Perform dot product with a given vector.
+   * VectorType::operator*=(const VectorType &);
+   *
+   *                // Scale all elements of the current vector and globally
+   *                // add a given vector to it.
+   * VectorType::add(const VectorType::value_type, const VectorType &);
+   *
+   *                      // Copies the data of a given vector to the current.
+   *                      // Resize the current vector if necessary (MPI safe).
+   * VectorType::operation=(const VectorType &);
+   *
+   *                 // To query the global size of the current vector.
+   * VectorType::size();
+   *
+   *                    // To query L_2 norm of the current vector
+   * VectorType::l2_norm();
+   *
+   *                  // Iterator to the start of the (locally owned) element
+   *                  // of the current vector.
+   * VectorType::begin();
+   *
+   *                // Iterator to the one past the last (locally owned)
+   *                // element of the current vector.
+   * VectorType::end();
+   *
+   *                     // Compress the vector i.e., flush the buffers of the
+   *                     // vector object if it has any.
+   * VectorType::compress(VectorOperation::insert);
+   * @endcode
+   *
+   * Most of the vectors in deal.II (see Vector classes) adhere to the above
+   * requirements.
+   *
+   * The current implementation in ROL doesn't support vector sizes above
+   * the largest value of int type.
+   */
+  template<typename VectorType>
+  class VectorAdaptor : public ROL::Vector<typename VectorType::value_type>
+  {
+
+    /**
+     * A typedef for size type of VectorType.
+     */
+    using size_type = typename VectorType::size_type;
+
+    /**
+     * A typedef for element type stored in the VectorType.
+     */
+    using value_type = typename VectorType::value_type;
+
+    /**
+     * A typedef for real-valued numbers.
+     */
+    using real_type = typename VectorType::real_type;
+
+  private:
+
+    /**
+     * Teuchos smart reference counting pointer to the Vector.
+     */
+    Teuchos::RCP<VectorType> vector_ptr;
+
+  public:
+
+    /**
+     * Constructor.
+     */
+    VectorAdaptor (const Teuchos::RCP<VectorType> &vector_ptr);
+
+    /**
+     * Returns the Teuchos smart reference counting pointer to the Vector,
+     * #vector_ptr.
+     */
+    Teuchos::RCP<VectorType> getVector ();
+
+    /**
+     * Returns the Teuchos smart reference counting pointer to const Vector.
+     */
+    Teuchos::RCP<const VectorType> getVector () const;
+
+    /**
+     * Return the dimension of the Vector.
+     */
+    int dimension () const;
+
+    /**
+     * Set the Vector to a given ROL::Vector @p rol_vector by overwriting its
```

Comment:
I think that each time you write "Vector" with a capital "V" then doxygen will link to our `Vector` class. I don't think that this is what you want, so I think that here and below you should change "Vector" to "vector" (or "wrapped vector"?).

Reply:
indeed, it will hyperlink it.